helm/polaris/values.yaml

```diff
@@ -0,0 +1,370 @@
+tolerations: []
+#  - key: "node-role.kubernetes.io/control-plane"
+#    operator: "Exists"
+#    effect: "NoSchedule"
+
+# -- Affinity and anti-affinity for polaris pods. See https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity.
+affinity: {}
+#  podAffinity:
+#    preferredDuringSchedulingIgnoredDuringExecution:
+#      - weight: 100
+#        podAffinityTerm:
+#          topologyKey: kubernetes.io/hostname
+#          labelSelector:
+#            matchExpressions:
+#              - key: app.kubernetes.io/name
+#                operator: In
+#                values:
+#                  - polaris
+
+# -- Configures the liveness probe for polaris pods.
+livenessProbe:
+  # -- Number of seconds after the container has started before liveness probes are initiated. Minimum value is 0.
+  initialDelaySeconds: 5
+  # -- How often (in seconds) to perform the probe. Minimum value is 1.
+  periodSeconds: 10
+  # -- Minimum consecutive successes for the probe to be considered successful after having failed. Minimum value is 1.
+  successThreshold: 1
+  # -- Minimum consecutive failures for the probe to be considered failed after having succeeded. Minimum value is 1.
+  failureThreshold: 3
+  # -- Number of seconds after which the probe times out. Minimum value is 1.
+  timeoutSeconds: 10
+  # -- Optional duration in seconds the pod needs to terminate gracefully upon probe failure. Minimum value is 1.
+  terminationGracePeriodSeconds: 30
+
+# -- Configures the readiness probe for polaris pods.
+readinessProbe:
+  # -- Number of seconds after the container has started before readiness probes are initiated. Minimum value is 0.
+  initialDelaySeconds: 5
+  # -- How often (in seconds) to perform the probe. Minimum value is 1.
+  periodSeconds: 10
+  # -- Minimum consecutive successes for the probe to be considered successful after having failed. Minimum value is 1.
+  successThreshold: 1
+  # -- Minimum consecutive failures for the probe to be considered failed after having succeeded. Minimum value is 1.
+  failureThreshold: 3
+  # -- Number of seconds after which the probe times out. Minimum value is 1.
+  timeoutSeconds: 10
+
+# -- Advanced configuration via Environment Variables.
+# Extra environment variables to add to the Polaris server container.
+# You can pass here any valid EnvVar object:
+# https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.27/#envvar-v1-core
+# This can be useful to get configuration values from Kubernetes secrets or config maps.
+extraEnv:
+  []
+#  - name: AWS_STORAGE_BUCKET
+#    value: s3://xxxxx/
+#  - name: AWS_ACCESS_KEY_ID
+#    valueFrom:
+#      secretKeyRef:
+#        name: aws-secret
+#        key: access_key_id
+#  - name: AWS_SECRET_ACCESS_KEY
+#    valueFrom:
+#      secretKeyRef:
+#        name: aws-secret
+#        key: secret_access_key
+
+# -- Configures for persistence.xml (ensure the key name is 'persistence.xml')
```

Comment:
nit: putting the entire `persistence.xml` into a secret looks like an overkill to me, but we can refactor that later.

Please add README instructions for creating this secret.

Reply:
Yes on the documentation part. Unfortunally, that is the only possible way as of now with EclipseLink unless we want to modify code to pull those auth info from other places such as env variables etc for the properties obj. See details in https://github.com/polaris-catalog/polaris/pull/88

Reply:
I think ideally the k8s secret should only contain values for `jakarta.persistence.jdbc.user` and `jakarta.persistence.jdbc.password`. Could we generate `persistence.xml` from those secrets (passed in as env. variables) in the init container?

For example: `sed -e "s/@@PASSWORD@@/$PASSWORD_FROM_K8S_SECRET/" <persistence.xml`

We'd have to load a copy of `persistence.xml` in helm and put the `@@PASSWORD@@` placeholder in the appropriate place.

Reply:
I am not very familiar with eclipse link and not sure if that is the only way for people to load username/password (e.g. what if the cert or another secret token is needed for connecting to backend) thus left it as an entire config in the jar without doing substitution. Then technically, we can populate the template for `persistence.xml` in `.values.yaml` then give instructions to user on what to set (e.g. `@@PASSWORD@@`) and behinds those onto the config and does the substitution. By doing so, we will not be limited to only allow the substitution of `URL, USERNAME, and PASSWORD`. What do you think?

Reply:
@MonkeyCanCode : I very much appreciate your work on starting a helm chart for Polaris!.. Still, it feels like the user experience is pretty complicated at this stage. As I commented in another thread, I think it would be nice to make some tooling first to simplify managing Polaris based on k8s-native configuration (simple config values or ConfgMap) and then revisit helm charts. I think that would be more manageable for developers and have better UX. WDYT?

Reply:
Alternatively, we could limit the scope of helm charts to in-memory operation initially (this PR) and add Eclipselink later.. WDYT?

Reply:
@dimas-b anytime. Yes, I do think the EclipseLink is really a mess at this stage and additional tooling for sure will be more helpful. I am open to ideas on how we should approach this. 

Then for limiting the scope, the default value is in-memory and even if users tried to set to EclipseLink, it wont work as there is no driver loaded in the gradle (update of code is needed). However, for advance user, this will be useable IMO as it is really 2-3 lines change and the service will be up and running in their k8s setup with proper persistent backend enabled.

Reply:
some ideas (beyond the scope of this PRs): [Quarkus](https://quarkus.io/) works pretty well for us in Nessie. It supports lots of backends and has k8s-friendly config.

Reply:
Yes, here is a similar discussion: https://github.com/polaris-catalog/polaris/discussions/125